tests/api_fastapi/core_api/routes/public/test_dag_run.py

```diff
@@ -254,3 +254,40 @@ def test_delete_dag_run_not_found(self, test_client):
         assert response.status_code == 404
         body = response.json()
         assert body["detail"] == "The DagRun with dag_id: `test_dag1` and run_id: `invalid` was not found"
+
+
+class TestClearDagRun:
+    def test_clear_dag_run(self, test_client):
+        response = test_client.post(
+            f"/public/dags/{DAG1_ID}/dagRuns/{DAG1_RUN1_ID}/clear", json={"dry_run": False}
+        )
+        assert response.status_code == 200
+        body = response.json()
+        assert body["dag_id"] == DAG1_ID
+        assert body["run_id"] == DAG1_RUN1_ID
+        assert body["state"] == "queued"
+
+    @pytest.mark.parametrize(
+        "body",
+        [{"dry_run": True}, {}],
+    )
+    def test_clear_dag_run_dry_run(self, test_client, body):
+        response = test_client.post(f"/public/dags/{DAG1_ID}/dagRuns/{DAG1_RUN1_ID}/clear", json=body)
+        assert response.status_code == 200
+        body = response.json()
+        assert body["total_entries"] == 1
+        for each in body["task_instances"]:
+            assert each["state"] is None
```

Comment:
Shouldn't the state still be `success` of the TI ? Before cleaning it's in success and after the dry_run it remains in success ?

Reply:
And I think we can assert the `DAG1_RUN1` state to be sure that nothing changed. (fetch from db like in the legacy test).

Reply:
Yeah. You are right. For some reason, dag.clear is returning task instance with None state. I'm looking into it.